relay-common/src/glob.rs

```diff
@@ -65,36 +65,38 @@ fn translate_codeowners_pattern(pattern: &str) -> Option<Regex> {
     // not deeper
     let trailing_slash_star = pattern.len() > 1 && pattern.ends_with("/*");
 
-    let mut iterator = pattern.chars().enumerate();
-
-    // Anchored paths may or may not start with a slash
-    if anchored && pattern.starts_with('/') {
-        iterator.next();
-        regex += r"/?";
-    }
+    let pattern_vec: Vec<char> = pattern.chars().collect();
 
     let mut num_to_skip = None;
-    for (i, ch) in iterator {
+    for (i, &ch) in pattern_vec.iter().enumerate() {
+        // Anchored paths may or may not start with a slash
+        if i == 0 && anchored && pattern.starts_with('/') {
+            regex += r"/?";
+            continue;
+        }
+
         if let Some(skip_amount) = num_to_skip {
             num_to_skip = Some(skip_amount - 1);
+            // Prevents everything after the * in the pattern from being skipped
             if num_to_skip > Some(0) {
                 continue;
             }
         }
+
         if ch == '*' {
             // Handle double star (**) case properly
-            if i + 1 < pattern.len() && pattern.chars().nth(i + 1) == Some('*') {
+            if i + 1 < pattern.len() && pattern_vec[i + 1] == '*' {
```

Comment:
```suggestion
            if pattern_vec.get(i + 1) == Some('*') {
```

See https://develop.sentry.dev/rust/#use-get-instead-of-slice-syntax.